From 4d0bf53af84f73c436544755bfdc11b276a4e432 Mon Sep 17 00:00:00 2001
From: Srujana Challa <schalla@marvell.com>
Date: Mon, 24 May 2021 17:19:30 +0530
Subject: [PATCH 1562/1921] crypto: cn10k: enable random number request.

Enable random number request(RNM_REQ_EN) to enable CPT to request
random numbers from RNM.

Signed-off-by: Srujana Challa <schalla@marvell.com>
Change-Id: I33fe1264010441e6f513091f42c5722aca8805ae
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52490
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/crypto/marvell/cn10k/cn10k_cptpf_ucode.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/marvell/cn10k/cn10k_cptpf_ucode.c b/drivers/crypto/marvell/cn10k/cn10k_cptpf_ucode.c
index 1fe5b760e7e1..54d83b43e9d5 100644
--- a/drivers/crypto/marvell/cn10k/cn10k_cptpf_ucode.c
+++ b/drivers/crypto/marvell/cn10k/cn10k_cptpf_ucode.c
@@ -2014,10 +2014,11 @@ int cn10k_cpt_try_create_default_eng_grps(struct pci_dev *pdev,
 	}
 	/*
 	 * Configure engine group mask to allow context prefetching
-	 * for the groups.
+	 * for the groups and enable random number request, to enable
+	 * CPT to request random numbers from RNM.
 	 */
 	cn10k_cpt_write_af_reg(pdev, CPT_AF_CTL,
-			       CN10K_CPT_ALL_ENG_GRPS_MASK << 3);
+			       CN10K_CPT_ALL_ENG_GRPS_MASK << 3 | BIT_ULL(16));
 	/*
 	 * Set interval to periodically flush dirty data for the next
 	 * CTX cache entry. Set the interval count to maximum supported
-- 
2.31.1

